### PR TITLE
First version of react frontend

### PR DESCRIPTION
**Pull request by tahoell**

### CODE REVIEW

File: react/.gitignore
Comment (hohwille) [inline react/.gitignore:6]: ich habe im Wurzelverzeichnis schon eine `.gitignore`. Da sind alle hidden files (`.*`) bereits ausgeschlossen:
https://github.com/devonfw-training/java-quarkus/blob/0312482f061e08c22f732ee2c86b859295ac4333/.gitignore#L1

Diese beiden Zeilen sind daher unnötig. Genauso auch 15-19
Auch `node_modules` würde ich auf oberster ebene ohne `/` hinzufügen.

**Review by hohwille — APPROVED** [review]
@tahoell thanks for this PR. It is really create that you managed to get a client up and running so fast. 👍 

I had a rough look but have to admit that I am reading `tsx` files for the first time and even TypeScript is not my skilled language.
Is there no boolean type for things like `completed`? It is currently `any` what IMHO is something like `Object` in Java.
I would have many other questions but I do not want to block this PR from being merged.
So I would vote to simply merge this and if we have something, we should rather create PRs for that.

File: react/.gitignore
Comment (tahoell) [reply react/.gitignore:6]: done

**Comment by tahoell** [comment]
> @tahoell thanks for this PR. It is really great that you managed to get a client up and running so fast. 👍
> 
> I had a rough look but have to admit that I am reading `tsx` files for the first time and even TypeScript is not my skilled language. Is there no boolean type for things like `completed`? It is currently `any` what IMHO is something like `Object` in Java. I would have many other questions but I do not want to block this PR from being merged. So I would vote to simply merge this and if we have something, we should rather create PRs for that.
Yes, there is a boolean type in TypeScript. Any is needed here for attributes `completed` and `starred` to do sorting operations in the function `MainContext#orderStarAndComplete`. That's why I left it at the moment as type `any`. 
I adjusted other remarks form you and will merge the PR then.

**Pull request merged**
